### PR TITLE
Jsdoc to markdown

### PR DESCRIPTION
**Pull request by hweeks**
### Summary:

This is a PR to move off of Jekyll and on to MkDocs. There is also a build file for auto generating MD files from our JSDoc. That is a to come feature, but part of this work still.

[The New Docs](https://hweeks.github.io/dynamoose/)

### MkDocs Help

[Building Locally](https://www.mkdocs.org/#building-the-site)

[Deploying Site](https://www.mkdocs.org/#deploying)

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [x] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoosejs/dynamoose/blob/master/LICENSE.txt)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have filled out all fields above

### CODE REVIEW

File: build-docs.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown')
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
const jsdoc2md = require('jsdoc-to-markdown');
```

File: build-docs.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown')
+const fs = require('fs')
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
const fs = require('fs');
```

File: build-docs.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown')
+const fs = require('fs')
+const glob = require('glob')
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
const glob = require('glob');
```

File: build-docs.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown')
+const fs = require('fs')
+const glob = require('glob')
+glob('lib/**/*.js', {}, (err, files) => {
+  const docs = files.map(file => {
+    const doc = jsdoc2md.renderSync({ files: file})
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
    const doc = jsdoc2md.renderSync({ files: file});
```

File: build-docs.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown')
+const fs = require('fs')
+const glob = require('glob')
+glob('lib/**/*.js', {}, (err, files) => {
+  const docs = files.map(file => {
+    const doc = jsdoc2md.renderSync({ files: file})
+    return {
+      location: `docs/${file.replace('.js', '.md')}`,
+      content: doc
+    }
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
    };
```

File: build-docs.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+const fs = require('fs')
+const glob = require('glob')
+glob('lib/**/*.js', {}, (err, files) => {
+  const docs = files.map(file => {
+    const doc = jsdoc2md.renderSync({ files: file})
+    return {
+      location: `docs/${file.replace('.js', '.md')}`,
+      content: doc
+    }
+  })
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
  });
```

File: build-docs.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  docs.forEach( docConfig => {
+    if (docConfig.content) {
+      fs.writeFileSync(docConfig.location, docConfig.content, 'utf8')
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
      fs.writeFileSync(docConfig.location, docConfig.content, 'utf8');
```

File: build-docs.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+const glob = require('glob')
+glob('lib/**/*.js', {}, (err, files) => {
+  const docs = files.map(file => {
+    const doc = jsdoc2md.renderSync({ files: file})
+    return {
+      location: `docs/${file.replace('.js', '.md')}`,
+      content: doc
+    }
+  })
+  docs.forEach( docConfig => {
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
  docs.forEach(docConfig => {
```

File: build-docs.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    if (docConfig.content) {
+      fs.writeFileSync(docConfig.location, docConfig.content, 'utf8')
+    }
+  })
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
  });
```

File: build-docs.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+})
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
});
```

File: docs/CNAME [hunk 11]
```diff
@@ -1 +1 @@
-dynamoosejs.com
\ No newline at end of file
+dynamoosejs.com
```
Comment (fishcharlie) [inline docs/CNAME]: Can we revert this change?

File: package.json [hunk 12]
```diff
@@ -65,7 +65,9 @@
     "cz-conventional-changelog": "2.1.0",
     "dynamodb-local": "0.0.24",
     "eslint": "5.13.0",
+    "glob": "^7.1.3",
```
Comment (fishcharlie) [inline package.json]:
```suggestion
    "glob": "7.1.3",
```

File: docs/javascripts/init.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
var jscode = document.getElementById('js-code');
```

File: docs/javascripts/init.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
+jscode.style.fontFamily = 'Roboto Mono'
+
+var bashcode = document.getElementById('bash-code')
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
var bashcode = document.getElementById('bash-code');
```

File: docs/javascripts/init.js [hunk 15]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
jscode.style.backgroundColor = '#f9fbfc';
```

File: docs/javascripts/init.js [hunk 16]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
jscode.style.fontSize = '14px';
```

File: docs/javascripts/init.js [hunk 17]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
+jscode.style.fontFamily = 'Roboto Mono'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
jscode.style.fontFamily = 'Roboto Mono';
```

File: docs/javascripts/init.js [hunk 18]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
+jscode.style.fontFamily = 'Roboto Mono'
+
+var bashcode = document.getElementById('bash-code')
+bashcode.style.backgroundColor = '#f9fbfc'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
bashcode.style.backgroundColor = '#f9fbfc';
```

File: docs/javascripts/init.js [hunk 19]
```diff
@@ -0,0 +1,12 @@
+hljs.initHighlightingOnLoad();
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
+jscode.style.fontFamily = 'Roboto Mono'
+
+var bashcode = document.getElementById('bash-code')
+bashcode.style.backgroundColor = '#f9fbfc'
+bashcode.style.fontSize = '14px'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
bashcode.style.fontSize = '14px';
```

File: docs/javascripts/init.js [hunk 20]
```diff
@@ -0,0 +1,12 @@
...
+
+var jscode = document.getElementById('js-code')
+jscode.style.backgroundColor = '#f9fbfc'
+jscode.style.fontSize = '14px'
+jscode.style.fontFamily = 'Roboto Mono'
+
+var bashcode = document.getElementById('bash-code')
+bashcode.style.backgroundColor = '#f9fbfc'
+bashcode.style.fontSize = '14px'
+bashcode.style.lineHeight = '30px'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
bashcode.style.lineHeight = '30px';
```

File: docs/javascripts/init.js [hunk 21]
```diff
@@ -0,0 +1,12 @@
...
+bashcode.style.fontFamily = 'Roboto Mono'
```
Comment (fishcharlie) [inline docs/javascripts/init.js]:
```suggestion
bashcode.style.fontFamily = 'Roboto Mono';
```

File: build-docs.js [hunk 22]
```diff
@@ -0,0 +1,17 @@
+const jsdoc2md = require('jsdoc-to-markdown');
+const fs = require('fs');
+const glob = require('glob');
+glob('lib/**/*.js', {}, (err, files) => {
+  const docs = files.map(file => {
+    const doc = jsdoc2md.renderSync({ files: file});
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
    const doc = jsdoc2md.renderSync({ files: file });
```

File: build-docs.js [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+    return {
+      location: `docs/${file.replace('.js', '.md')}`,
+      content: doc
+    };
+  });
+  docs.forEach( docConfig => {
```
Comment (fishcharlie) [inline build-docs.js]:
```suggestion
  docs.forEach(docConfig => {
```

**Comment by hweeks** [comment]
@fishcharlie i'll be updating to the new lint in a day or so. i'll also merge in the plugin stuff.

**Comment by hweeks** [comment]
this has been updated, a travis job has been added.

**Comment by hweeks** [comment]
For anyone that would like too see the deployed version: https://hweeks.github.io/dynamoose/

File: docs/CNAME (hunk 11 above)
Comment (fishcharlie) [reply docs/CNAME]: @hweeks

File: lib/Plugin.js [hunk 24]
```diff
@@ -3,6 +3,18 @@
 const debug = require('debug')('dynamoose:plugin');
 const Listener = require('./Listener');
 
+/**
+ * @alias module:plugin
+ * @typicalname Plguin
```
Comment (fishcharlie) [inline lib/Plugin.js]:
```suggestion
 * @typicalname Plugin
```

File: package.json (hunk 12 above)
Comment (fishcharlie) [reply package.json]: @hweeks

**Comment by fishcharlie** [comment]
@hweeks @rish-16 How did this get merged in? It looks like there are a lot of design problems just quickly glancing at it.

**Comment by hweeks** [comment]
what issues? mobile? i’ve just pulled the interior content of the main
page.

**Comment by fishcharlie** [comment]
@hweeks It just needs a detailed review. I see one typo. There is a weird border around the `export`'s. Did you ever fix the two comments I had in my last review?

**Comment by fishcharlie** [comment]
Haven't tested on mobile yet

**Comment by hweeks** [comment]
i did address your comments
